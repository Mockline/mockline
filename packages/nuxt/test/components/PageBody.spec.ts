import { describe, it, expect } from 'vitest'
import type {TypeOf} from "zod";
//@ts-ignore
import { MPageBody } from '#components';
import ComponentRender from "../component-render";

describe('PageBody', () => {
  it.each([
    ['base case', {}],
    ['<MPageBody class="p-1" />']
    // @ts-expect-error
  ])('renders %s correctly', async (nameOrHtml, options: TypeOf<typeof MPageBody.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MPageBody)
    expect(html).toMatchSnapshot()
  })
})
