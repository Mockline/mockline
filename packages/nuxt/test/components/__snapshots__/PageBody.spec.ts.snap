// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageBody > renders <MPageBody class="p-1" /> correctly 1`] = `"<div class="flex flex-col gap-4 w-full p-1"></div>"`;

exports[`PageBody > renders base case correctly 1`] = `"<div class="flex flex-col gap-4 w-full"></div>"`;
